src/ir_def/construct.ml

```diff
@@ -91,6 +91,7 @@ let primE prim es =
     | ICStableWrite _ -> T.unit
     | IcUrlOfBlob -> T.text
     | ActorOfIdBlob t -> t
+    | BinPrim (t, _) -> t
```

Comment:
Well, it is back, because this code is now invoke due to building IR by `primE`, and I am generating addition nodes in IR now. We get an assertion failure when taking this out:
```
Fatal error: exception "Assert_failure ir_def/construct.ml:99:11"
Raised at Ir_def__Construct.primE in file "ir_def/construct.ml", line 99, characters 11-23
Called from Lowering__Desugar.sequentialForE.body in file "lowering/desugar.ml", line 353, characters 15-146
Called from Lowering__Desugar.sequentialForE in file "lowering/desugar.ml", line 361, characters 11-89
Called from Lowering__Desugar.exp' in file "lowering/desugar.ml", line 204, characters 147-202
Called from Lowering__Desugar.typed_phrase' in file "lowering/desugar.ml", line 45, characters 16-30
Called from Lowering__Desugar.dec' in file "lowering/desugar.ml", line 616, characters 22-29
Called from Lowering__Desugar.phrase' in file "lowering/desugar.ml", line 41, characters 16-34
Called from Lowering__Desugar.dec in file "lowering/desugar.ml", line 613, characters 14-30
Called from Stdlib__list.map in file "list.ml", line 92, characters 20-23
Called from Lowering__Desugar.transform_unit_body in file "lowering/desugar.ml", line 936, characters 26-35
Called from Lowering__Desugar.transform_unit in file "lowering/desugar.ml", line 974, characters 14-38
Called from Pipeline.desugar_unit in file "pipeline/pipeline.ml", line 542, characters 4-22
Called from Pipeline.compile_unit in file "pipeline/pipeline.ml", line 626, characters 16-43
Called from Pipeline.compile_files in file "pipeline/pipeline.ml", line 647, characters 14-53
Called from Diag.bind in file "lang_utils/diag.ml", line 32, characters 27-30
Called from Diag.bind in file "lang_utils/diag.ml", line 32, characters 27-30
Called from Moc.process_files in file "exes/moc.ml", line 152, characters 37-82
Called from Moc in file "exes/moc.ml", line 214, characters 4-23
```
Nothing nefarious going on here.

Reply:
Yes, my comment was misguided, I thought you were introducing a `BinE`; `BinPrim` is fine.